### PR TITLE
Add top-level Visualize data TOC section.

### LINKED ISSUE #4510: Docs: Create Visualize data top-level section

### PR DESCRIPTION
**Pull request by osg-grafana**
Fix #4510

### CODE REVIEW

File: docs/sources/mimir/visualize/grafana.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+weight: 100
+keywords:
+  - visualize
+  - grafana
+  - dashboards
+---
+
+# Visualize metric data
+
+[Grafana](/grafana/download/) has built-in support for Prometheus.
```
Comment (charleskorn) [inline docs/sources/mimir/visualize/grafana.md:14]: Should we rephrase this to be clear that the support is for Mimir?

```suggestion
[Grafana](/grafana/download/) has built-in support for Mimir through the Prometheus data source type.
```

File: docs/sources/mimir/visualize/grafana.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+   - When running locally or with Docker using port mapping,
+     the address is likely `http://localhost:9090`.
+
+   - When running with docker-compose or Kubernetes,
+     the address is likely `http://prometheus:9090`.
+
+1. To see metrics, select **Explore**.
+1. Select a metric, apply (optional) label filters, and select **Run query**.
+1. Learn more about PromQL and querying by reading [Querying Prometheus](https://prometheus.io/docs/prometheus/latest/querying/basics/).
```
[the remark this answers is not in the document]
Comment (charleskorn) [reply docs/sources/mimir/visualize/grafana.md]: Sure. I think it might still be beneficial to split the steps here (the first four are once-off steps, the last three steps are the steps you need to follow each time).

[the remark this answers is not in the document]
Comment (osg-grafana) [reply docs/sources/mimir/visualize/grafana.md]: I have attempted to split these two procedures into two.

File: docs/sources/mimir/visualize/grafana.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+   the username and password are both `admin`.
+1. From within Grafana, go to `Connections` > `Add new connection`.
+1. Search for Prometheus, and select **Create a Prometheus data source**.
+1. In the **HTTP** > **Prometheus server URL** field, point to the proxy in front of Mimir (such as Nginx or the GEM gateway) rather than to one component of Mimir.
+
+   - When running locally or with Docker using port mapping,
+     the address is likely `http://localhost:9090`.
+
+   - When running with docker-compose or Kubernetes,
+     the address is likely `http://prometheus:9090`.
```
Comment (charleskorn) [inline docs/sources/mimir/visualize/grafana.md:29]: I think we need to talk about this in terms of the tool used to install Mimir (Helm, Jsonnet etc.) rather than the environment Mimir is running in (local or Kubernetes), as each installation method has different configuration, and that then determines the URL.

For Helm, based on https://grafana.com/docs/helm-charts/mimir-distributed/latest/get-started-helm-charts/#start-grafana-in-kubernetes-and-query-metrics, the default URL to use would be `http://mimir-nginx.<Mimir namespace>.svc/prometheus`.

For Jsonnet/Tanka, we don't install a proxy automatically (probably something we could improve). 

If you're running the binary locally in monolithic mode, you can use `http://localhost:<Mimir port>`. If you're running microservices, then you need a proxy.

File: docs/sources/mimir/visualize/grafana.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+## Steps
+
+1. To see metrics, select **Explore**.
```
Comment (JStickler) [inline docs/sources/mimir/visualize/grafana.md:33]: If the user has already completed the "before you begin" steps, they're going to skip right down to this part.  Maybe update this to provide context for where they should be.
```suggestion
1. To view metrics in Grafana, select **Explore**.
```

File: docs/sources/mimir/visualize/grafana.md (hunk 3 above)
Comment (osg-grafana) [reply docs/sources/mimir/visualize/grafana.md]: Please have another look.

File: docs/sources/mimir/visualize/grafana.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+1. Log in to your Grafana instance.
+   If this is your first time running Grafana,
+   the username and password are both `admin`.
+1. From within Grafana, go to `Connections` > `Add new connection`.
+1. Search for Prometheus, and select **Create a Prometheus data source**.
+1. In the **HTTP** > **Prometheus server URL** field, enter a server URL:
+
+   - If you deployed Mimir via the `mimir-distributed` Helm chart,
+     the default URL is is `http://mimir-nginx.<MIMIR-NAMESPACE>.svc/prometheus`.
```
Comment (krajorama) [inline docs/sources/mimir/visualize/grafana.md:26]:
```suggestion
     the default URL inside the cluster is is `http://<HELM_RELEASE>-nginx.<MIMIR-NAMESPACE>.svc/prometheus`,
     and `http://<INGRESS-HOST>/prometheus` from outside, provided that you set up an ingress.
```

File: docs/sources/mimir/visualize/grafana.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+   - If you deployed Mimir via Jsonnet and Tanka,
+     use `http://localhost:<MIMIR-PORT>`.
```
Comment (krajorama) [inline docs/sources/mimir/visualize/grafana.md:29]: I assume you need the `/prometheus` suffix here as well. But I don't use tanka, maybe ask @pstibrany

File: docs/sources/mimir/visualize/grafana.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+1. Log in to your Grafana instance.
+   If this is your first time running Grafana,
+   the username and password are both `admin`.
+1. From within Grafana, go to `Connections` > `Add new connection`.
+1. Search for Prometheus, and select **Create a Prometheus data source**.
+1. In the **HTTP** > **Prometheus server URL** field, enter a server URL:
+
+   - If you deployed Mimir via the `mimir-distributed` Helm chart,
+     the default URL inside the cluster is `http://<HELM-RELEASE>-nginx.<MIMIR-NAMESPACE>.svc/prometheus`,
+     and `http://<INGRESS-HOST>/prometheus` from outside, provided that you set up an ingress.
```
Comment (krajorama) [inline docs/sources/mimir/visualize/grafana.md:27]:
```suggestion
     the default URL inside the Kubernetes cluster is `http://<HELM-RELEASE>-nginx.<MIMIR-NAMESPACE>.svc/prometheus`,
     and `http://<INGRESS-HOST>/prometheus` from the outside, provided that you set up an ingress.
```

File: docs/sources/mimir/visualize/grafana.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+## Before you begin
+
+1. Log in to your Grafana instance.
+   If this is your first time running Grafana,
+   the username and password are both `admin`.
+1. From within Grafana, go to `Connections` > `Add new connection`.
+1. Search for Prometheus, and select **Create a Prometheus data source**.
+1. In the **HTTP** > **Prometheus server URL** field, enter a server URL:
+
+   - If you deployed Mimir via the `mimir-distributed` Helm chart,
```
Comment (krajorama) [inline docs/sources/mimir/visualize/grafana.md:25]: I would also mention that both `helm install` and `helm upgrade` print the actual URLs to use for reading/writing to Mimir.

File: docs/sources/mimir/visualize/grafana.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+   - If you deployed Mimir via Jsonnet and Tanka,
+     use `http://localhost:<MIMIR-PORT>`.
+
+   - If you are running microservices, point to the proxy in front of Mimir
+     (such as Nginx or the GEM gateway) rather than to one component of Mimir.
+
+## Steps
+
+1. To view metrics in Grafana, select **Explore**.
```
Comment (krajorama) [inline docs/sources/mimir/visualize/grafana.md:37]: After going to Explore, you also need to select the newly created data source at the top left.

File: docs/sources/mimir/visualize/grafana.md (hunk 8 above)
Comment (osg-grafana) [reply docs/sources/mimir/visualize/grafana.md:25]: Thank you. This is super helpful for the user.

File: docs/sources/mimir/visualize/grafana.md (hunk 6 above)
Comment (osg-grafana) [reply docs/sources/mimir/visualize/grafana.md:30]: I could move this to a follow-on issue to iterate more quickly. We can always add it back in, what do you think?

Comment (krajorama) [reply docs/sources/mimir/visualize/grafana.md:30]:
```suggestion
     use `http://localhost:<MIMIR-PORT>/prometheus`.
```

Comment (krajorama) [reply docs/sources/mimir/visualize/grafana.md:30]: actually jsonnet/tanka is for kubernetes as well, so unless you do a port forward to localhost, this is not true

Comment (pstibrany) [reply docs/sources/mimir/visualize/grafana.md:30]: We don't override `-http.prometheus-http-prefix` flag in Jsonnet, so it defaults to `/prometheus` there as well.

**Review by krajorama — APPROVED** [review]
LGTM as a starting base
